### PR TITLE
[NCL-6402] Add endpoint for build configs that include latest build

### PR DESCRIPTION
**Pull request by dominikbrazdil**
### Checklist:

* [x] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProviderImpl.java [hunk 1]
```diff
@@ -320,6 +325,47 @@ public BuildConfigurationRevision createRevision(String id, BuildConfiguration b
...
+            String sortingRsql,
+            String query) {
+        Page<BuildConfiguration> buildConfigurations = queryForCollection(
+                pageIndex,
+                pageSize,
+                sortingRsql,
+                query,
+                isNotArchived());
+        List<BuildConfigurationWithLatest> bcsWithLatest = new ArrayList<>();
+        buildConfigurations.getContent().forEach(bc -> {
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProviderImpl.java:341]: I would maybe extract the lambda to a method. This method is getting a little bit long.
probalby `BuildConfigurationWithLatest someGoodName(BuildConfiguration buildConfig)`

**Review by janinko — CHANGES_REQUESTED** [review]

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java [hunk 2]
```diff
@@ -117,6 +119,32 @@
...
+                    @ApiResponse(
+                            responseCode = INVALID_CODE,
+                            description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    Page<BuildConfigurationWithLatest> getAllWithLatestBuild(@Valid @BeanParam PageParameters pageParams);
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java:146]: Is there `@Path` annotation missing?

Comment (dominikbrazdil) [reply rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java:146]: Yeah.. I'm stupid

**Comment by dominikbrazdil** [comment]
retest this please

**Review by janinko — APPROVED** [review]

**Pull request merged**